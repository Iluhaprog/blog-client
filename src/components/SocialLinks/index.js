import sl from './SocialLinks';
import { connect } from 'react-redux';
import { 
    createSocialLinkFetch,
    updateSocialLinkFetch,
    updateSocialLinkPreviewFetch,
} from '../../actoins/socialLink';
import { addError } from '../../actoins/error';
import { getUniqueName } from '../../util/string/string';
import { setErrorCatch } from '../../util/SettingErrorCatch';

const mapStateToProps = state => ({
    socials: state.socialLink.socialLinks,
});

const mapDispatchToProps = dispatch => ({
    update: socialLink => {
        dispatch(updateSocialLinkFetch(socialLink))
    },
    updatePreview: (id, file) => {
        if (file) {
            const fd = new FormData();
            fd.append('file', new File([file], getUniqueName(file.name)));
            setErrorCatch(
                dispatch(updateSocialLinkPreviewFetch(id, fd)),
                e => {
                    dispatch(addError(e));
                }
            );
        }
    },
    create: socialLink => {
        dispatch(createSocialLinkFetch(socialLink));
    },
});

const SocialLinks = connect(mapStateToProps, mapDispatchToProps)(sl);

export {
    SocialLinks,
};