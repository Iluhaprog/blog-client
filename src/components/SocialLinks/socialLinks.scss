@import '../../vars.scss';

.social-links {
    max-width: 650px;

    .drop-zone {
        width: 60px;
        height: 60px;
        background-color: $text-color-6;

        img.empty {
            width: 40px;
            height: auto;
            margin-top: 12px;
        }
    }
}