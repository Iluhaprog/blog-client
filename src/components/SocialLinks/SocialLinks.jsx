import React from 'react';
import PropTypes from 'prop-types';
import { LabeledButton } from '../buttons';
import { Row, Column } from '../containers';
import { SocialLinkForm } from '../forms';
import './socialLinks.scss';

function SocialLinks(props) {
    const { create, update, updatePreview } = props;
    return (
        <div className='social-links'>
            <Column>
                <Row justifyContent='sb' alignItems='c'>
                    <h3>
                        Socials
                    </h3>
                    <LabeledButton
                        text='Add'
                        onClick={() => {
                            create({ 
                                title: '',
                                preview: '',
                                link: '',
                            });
                        }} 
                    />
                </Row>
                {
                    props.socials.map(social => (
                        <SocialLinkForm 
                            form={'social' + social.id}
                            onSubmit={values => {
                                update({ 
                                    ...social,
                                    title: values.socialTitle,
                                    link: values.socialLink,
                                });
                                if (values.preview) {
                                    updatePreview(social.id, values.preview);
                                }
                            }}
                            key={social.id}
                            id={social.id}
                            preview={social.preview}
                            initialValues={{
                                socialTitle: social.title,
                                socialLink: social.link,
                            }}
                        />
                    ))
                }
            </Column>
        </div>
    );
}

SocialLinks.defaultProps = {
    socials: [],
};

SocialLinks.propTypes = {
    socials: PropTypes.array,
};

export default SocialLinks;