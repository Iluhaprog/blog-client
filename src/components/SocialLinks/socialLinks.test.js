import React from  'react';
import { render } from '@testing-library/react';
import SocialLinks from './SocialLinks';

test('test view of social links components', () => {
    const { asFragment } = render(<SocialLinks />);
    expect(asFragment()).toMatchSnapshot();
});