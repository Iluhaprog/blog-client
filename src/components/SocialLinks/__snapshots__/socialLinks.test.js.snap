// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test view of social links components 1`] = `
<DocumentFragment>
  <div
    class="social-links"
  >
    <div
      class="column   "
    >
      <div
        class="row row_jc-sb row_ai-c "
      >
        <h3
          class="social-links__title"
        >
          Socials
        </h3>
        <div
          class="row row_jc-fe  "
        >
          <div
            class="loader "
          >
            <img
              alt="loading..."
              height="30"
              src="loader.svg"
              width="30"
            />
          </div>
          <button
            class="button default-button labeled-button"
          >
            <div
              class="row   "
            >
              <p>
                <span />
                <span />
              </p>
              <p>
                Add
              </p>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
